substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -771,29 +776,29 @@ impl<T: Config> Module<T> {
         Ok(contract)
     }
 
-    fn handle_lock(contract: &mut types::Contract, amount_due: BalanceOf<T>) {
+    fn handle_lock(contract: &mut types::Contract, amount_due: BalanceOf<T>) -> DispatchResult {
         let now = <timestamp::Module<T>>::get().saturated_into::<u64>() / 1000;
-        let mut contract_lock = ContractLock::<T>::get(contract.contract_id);
-        let new_amount_locked = contract_lock.amount_locked + amount_due;
 
         // increment cycles billed and update the internal lock struct
+        let mut contract_lock = ContractLock::<T>::get(contract.contract_id);
         contract_lock.lock_updated = now;
         contract_lock.cycles += 1;
-        contract_lock.amount_locked = new_amount_locked;
+        contract_lock.amount_locked = contract_lock.amount_locked + amount_due;
         ContractLock::<T>::insert(contract.contract_id, &contract_lock);
 
         // Contract is in grace state, don't actually lock tokens or distribute rewards
         if matches!(contract.state, types::ContractState::GracePeriod(_)) {
-            return;
+            return Ok(())
         }
 
-        let twin = pallet_tfgrid::Twins::<T>::get(contract.twin_id);
         // Only lock an amount from the user's balance if the contract is in create state
-        // Update lock for contract and ContractLock in storage
+        // The lock is specified on the user's account, since a user can have multiple contracts
+        // Just extend the lock with the amount due for this contract billing period (lock will be created if not exists)
+        let twin = pallet_tfgrid::Twins::<T>::get(contract.twin_id);
         <T as Config>::Currency::extend_lock(
-            contract.contract_id.to_be_bytes(),
+            GRID_LOCK_ID,
             &twin.account_id,
-            new_amount_locked.into(),
+            amount_due,
             WithdrawReasons::RESERVE,
```

Comment:
Lets change this to `WithdrawRewasons::all()` so we are sure locked funds can not be moved by anything